test/ussd_clinic.test.js

```diff
@@ -574,6 +574,33 @@ describe("app", function() {
             });
         });
 
+        describe("when on state_mobile_no and number has no subscription", function() {
+            it("should go to state_consent", function() {
+                return tester
+                .setup(function(api) {
+                    api.http.fixtures.add(fixtures_IdentityStoreDynamic.identity_search({
+                        'msisdn': '+27820001001'
+                    }));
+                    api.http.fixtures.add(fixtures_StageBasedMessagingDynamic.active_subscriptions());
+                })
+                .setup.user.answer('operator', {'id': 'operator-id'})
+                .setup.user.answer('operator_msisdn', '+27820001002')
+                .setup.user.state('state_mobile_no')
+                .input('0820001001')
+                .check.interaction({
+                    state: "state_consent",
+                    reply: [
+                            'We need to collect, store & use her info. She ' +
+                            'may get messages on public holidays & weekends. ' +
+                            'Does she consent?',
+                            '1. Yes',
+                            '2. No'
+                    ].join('\n')
+                })
+                .run();
+            });
+        });
```

Comment:
I don't think that this test will trigger the bug. We need to test that the output of `state_already_subscribed` shows the entered number, not the current handset's number.

We can probably modify one of the existing tests to have a different `setup.user.addr` to what's configured in `registrant_msisdn`, and ensure that the response includes `registrant_msisdn`

Reply:
Thanks will test for that as well